### DISCUSSION TITLE
[Feature request] Priority queue support

### DISCUSSION THREAD

**Discussion by approached** [postalserver/postal#1491] [Feature suggestions]
Hi @ all

Me need a priority queue support for sending emails. For example:
I start a newsletter for 20.000 recipients to send. In the background i have a website with transaction emails (like for registration or contact email).

It would be nice if i can say, transaction emails have more priority.

**Comment by mjnaderi** [comment]
Any updates on this? This is really essential for us.

**Reply by mjnaderi** [reply]
I created two mail servers in the same organization: one for "normal" messages and one for "high priority" messages. My assumption was that each mail server has a separate queue, so this helps high priority messages to be processed independently.

But it did not help. When there are lots of messages in the queue of "normal" mail server, the emails in "high priority" mail server get stuck for long time.

**Reply by willpower232** [reply]
Unfortunately there is only one queue for the whole of Postal regardless of how many servers you make.

**Reply by mjnaderi** [reply]
Thanks for the clarification. @willpower232

Here's the solution I'm thinking of:
We should apply throttling for normal emails in our own app's email queue (and have no throttling for high priority emails). I'm thinking of 2 api calls per second to avoid congesting Postal's queue and let high priority emails go through instantly.

**Reply by willpower232** [reply]
that makes sense, unfortunately the solution is to slow down the outgoing messages from whatever puts them into Postal, Postal mostly works for transactional emails so doesn't have complicated marketing realities in mind

**Reply by notz** [reply]
@adamcooke No chance to implement a priority flag or int value per server and use that in the queue? You recently changed the whole queuing implementation.

**Reply by adamcooke** [reply]
I don't hate the idea!

**Reply by mjnaderi** [reply]
Hi @adamcooke,

We’ve implemented the idea and opened this pull request: https://github.com/postalserver/postal/pull/3446.
Could you please review and consider merging it?

The absence of the "priority" feature has caused us significant issues, and since the change includes a database migration to add the "priority" field, we’d prefer to stay aligned with the official image rather than diverging our database from the official project and maintaining a custom one.

Thank you!
